### PR TITLE
Design review dashboard

### LINKED ISSUE #2020: Design Review: Dashboard Menu
**REALLY IMPORTANT**
- [x] wording : My Titles (not just "Titles"
image

**If it's possible** 
-  [x] remove divider
image
- [x] Add an active statu when you are on a page (like exemple) 
image

### LINKED ISSUE #2024: Design Review: Offers and Deals
**REALLY IMPORTANT**
- [x] change button for fab button (like others pages)
image
- [x] 16px margin between them
image
- [x] active tabs (primary)
image
- [x] margin = 8px between filter and table
image

### PR DESCRIPTION
**Pull request by Mussuranae**
Review of the dashboard "offers and deals page" and sidenav menu
Fix #2020 and fix #2024

### CODE REVIEW

File: libs/ui/src/lib/list/table-filter/table-filter.component.scss [hunk 1]
```diff
@@ -10,6 +11,21 @@ table {
   width: 100%;
 }
 
+.icon-filter {
+  margin-top: 8px;
+  &.first-icon {
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/list/table-filter/table-filter.component.scss]: consider using native css semantic when available : 
```css
.icon-filter:first-child {
  margin-right: 8px;
}
```
Here is the list : https://www.w3schools.com/cssref/css_selectors.asp

File: libs/ui/src/lib/list/table-filter/table-filter.component.scss [hunk 2]
```diff
@@ -10,6 +11,21 @@ table {
...
+    margin-right: 8px;
+  }
+}
+
+.padding-filter {
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/list/table-filter/table-filter.component.scss]: Select with "article" if possible to apply the padding.

File: libs/utils/src/lib/algolia-autocomplete/algolia-autocomplete.component.ts [hunk 3]
```diff
@@ -52,7 +52,7 @@ export class AlgoliaAutocompleteComponent implements OnInit {
   /**
    * Output to get all data from algolia
    */
-  @Output() emitObjectID = new EventEmitter();
+  @Output() emitSelect = new EventEmitter();
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/algolia-autocomplete/algolia-autocomplete.component.ts]: what about `selectionChange` to match with the mat-select output event ?

File: styles/components/bf-table.scss [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
-      // Doesn't work. We need to find the right class
-      .mat-form-field-outline, input {
-        background-color: mat-color($background, background);
-      }
+  bf-table-filter {
+    // Doesn't work. We need to find the right class
+    .mat-form-field-outline, input {
+      background-color: mat-color($background, background);
+    }
+    mat-form-field .mat-form-field-wrapper {
```
Comment (GrandSchtroumpf) [inline styles/components/bf-table.scss]: I would I put that in the scss of the component as it's not related to the theme.

File: styles/components/mat-list.scss [hunk 5]
```diff
@@ -21,12 +21,10 @@
...
   .mat-nav-list a.active {
     color: mat-color($primary, 500);
-    &:focus {
-      background: transparent;
-    }
+    background-color: rgba(mat-color($primary, 500), 0.05);
 
     // Mat icons inside active link should be colored too
-    .mat-list-icon svg g {
+    .mat-list-icon path, .mat-list-icon g, .mat-list-icon polygon {
```
Comment (GrandSchtroumpf) [inline styles/components/mat-list.scss]: Maybe easier to read like that : 
```css
.mat-list-icon {
  path, g, polygon {
    fill: mat-color($primary, 500);
  }
}
```

File: styles/components/mat-sidenav.scss [hunk 6]
```diff
@@ -3,6 +3,11 @@
 
   mat-sidenav {
     width: 256px;
+
+    // No border bottom for the toolbar in the dashboard sidenav
+    &#dashboard-navbar .mat-toolbar {
```
Comment (GrandSchtroumpf) [inline styles/components/mat-sidenav.scss]: This should be in the dashboard layout component style.

File: styles/components/bf-table.scss (hunk 4 above)
Comment (Mussuranae) [reply styles/components/bf-table.scss]: I don't success to put it in the component :/ I try a lot of time in the parent component and in the component where I need the change, and nothing works

Comment (GrandSchtroumpf) [reply styles/components/bf-table.scss]: ok keep it here

**Pull request merged**